internal/manifest/version.go

```diff
@@ -256,31 +256,39 @@ func (v *Version) Next() *Version {
 // may touch). If level is zero then that assumption cannot be made, and the
 // [start, end] range is expanded to the union of those matching ranges so far
 // and the computation is repeated until [start, end] stabilizes.
+// The returned files are a subsequence of the input files, i.e., the ordering
+// is not changed.
 func (v *Version) Overlaps(
 	level int, cmp Compare, start, end []byte,
 ) (ret []FileMetadata) {
 	if level == 0 {
-		// The sstables in level 0 can overlap with each other. As soon as we find
-		// one sstable that overlaps with our target range, we need to expand the
-		// range and find all sstables that overlap with the expanded range.
-	loop:
+		// Indices that have been selected as overlapping.
+		selectedIndices := make([]bool, len(v.Files[level]))
+		numSelected := 0
+		restart := false
```

Comment:
You can move this line inside of the for-loop which will allow you to not need to set `restart = false` later on.

Reply:
done